src/main/java/software/amazon/nio/spi/s3/S3FileSystemProvider.java

```diff
@@ -821,4 +825,18 @@ protected Map<String, Object> readAttributes(S3AsyncClient client, Path path, St
     public void setAttribute(Path path, String attribute, Object value, LinkOption... options) throws UnsupportedOperationException {
         throw new UnsupportedOperationException("s3 file attributes cannot be modified by this class");
     }
+
+    /**
+     * Creates if needed and returns the S3ClientStrore caching all created
+     * clients. This is not public on purpose, as it is an intermediate solution
+     * that will be replaced by instance accessors.
+     *
+     * @return the chache of clients as a S3ClientStore
+     */
+    protected static S3ClientStore getClientStore() {
```

Comment:
If it is intended to be replaced, can it be private?

Reply:
It can not be private because it is called instead of S3ClientStore.getInstance() in many places outside the class.